pyDeltaRCM/sed_tools.py

```diff
@@ -151,24 +148,68 @@ def topo_diffusion(self):
             self.eta += self.cf
 
 
+@njit
+def _get_weight_at_cell_sediment(ind, weight_int, depth_nbrs, ct_nbrs,
+                                 dry_depth, theta, distances_flat):
+
+    dry = (depth_nbrs <= dry_depth)
+    wall = (ct_nbrs == -2)
+    ctr = (np.arange(9) == 4)
+    drywall = np.logical_or(dry, wall)
+
+    # always set ctr to 0 before rebalancing
+    weight_int[ctr] = 0
+
+    weight = np.copy(weight_int)  # no gamma weighting here
+    weight = (depth_nbrs ** theta) * weight
+
+    # ALWAYS disallow the choice to not move
+    weight[ctr] = 0
+
+    # ALWAYS disallow the choice to go into land or dry cells
+    weight[drywall] = 0
+
+    # sanity check
+    if np.any(np.isnan(weight)):
+        raise RuntimeError('NaN encountered in sediment weighting.'
+                           'Please report error.')
+
+    # correct the weights for random choice
+    if np.nansum(weight) == 0:
```

Comment:
So just to check, if there's a check to no longer have NaNs in the weighting arrays, should we just use regular `np.sum()`? I think it's wildly faster actually.

Reply:
yeah, good call. I made this change in the water weighting, but I guess I forgot to switch it for the sediments!